Master

randomdavis opened the pull request:
Adds job opening request/approval/denial/fetching functionality

randomdavis committed:
Adds necessary data memebers top opening entity and contract to allow opening requests to be made

randomdavis committed:
Adds store ID to job opening contract

randomdavis committed:
Adds job opening request funcitons to service, as well as unit tests (all pass)

randomdavis committed:
Merge remote-tracking branch 'refs/remotes/DavidA94/master'

randomdavis commented:
I know my tests have a lot of redundant code. Ignore that for now if you please.

DavidA94 commented on AES.OpeningsSvc/OpeningSvc.svc.cs:
```diff
@@ -15,19 +16,125 @@ public OpeningSvc()
+            return GetOpenings(StoreID, OpeningStatus.REJECTED);
+        }
+
+        public bool RequestOpening(int StoreID, JobOpeningContract opening)
```
How will n openings be requested? Will they have to call this n times?

randomdavis replied:
Fixed in my next push

DavidA94 commented on AES.OpeningsSvc/OpeningSvc.svc.cs:
```diff
@@ -15,19 +16,125 @@ public OpeningSvc()
+                };
+
+                db.JobOpenings.Add(openingRequest);
+                db.SaveChanges();
```
Good idea to ensure that this saves by ensuring the number returned is >0.

randomdavis replied:
Fixed in my next push

DavidA94 commented on AES.OpeningsSvc/OpeningSvc.svc.cs:
```diff
@@ -15,19 +16,125 @@ public OpeningSvc()
+                catch (System.Exception ex)
+                {
+
+                    throw ex;
```
Theoretically, the service should never throw an error. Plus, the catch -> throw makes this try/catch pointless. Either return false or remove the try.

randomdavis replied:
Removed t/c in next push

DavidA94 commented on AES.OpeningsSvc/OpeningSvc.svc.cs:
```diff
@@ -15,19 +16,125 @@ public OpeningSvc()
+            return returnedList;
+        }
 
+        private List<JobOpeningContract> GetOpenings(int StoreID, OpeningStatus status)
```
Good use of helper method.

randomdavis replied:
Thanks!

DavidA94 commented on Tests/AES.OpeningSvc.Tests/OpeningSvcUnitTest.cs:
```diff
@@ -126,5 +127,231 @@ public void OpeningSvc_GetOpenings()
+
+                db.Stores.RemoveRange(db.Stores.Where(st => st.ID == newStore.ID));
+
+                db.SaveChanges();
```
No point to hit the DB 3 times. RemoveAll -> Save.
Ditto on the other ones.

randomdavis replied:
I don't want to remove all openings, just the ones that match the filter I gave. Still, I removed the redundant saves in my next push.

DavidA94 commented on Tests/AES.OpeningSvc.Tests/OpeningSvcUnitTest.cs:
```diff
@@ -126,5 +127,231 @@ public void OpeningSvc_GetOpenings()
+
+                db.Stores.RemoveRange(db.Stores.Where(st => st.ID == newStore.ID));
+
+                db.SaveChanges();
```
Ditto.

DavidA94 commented on Tests/AES.OpeningSvc.Tests/OpeningSvcUnitTest.cs:
```diff
@@ -126,5 +127,231 @@ public void OpeningSvc_GetOpenings()
+
+                db.Stores.RemoveRange(db.Stores.Where(st => st.ID == newStore.ID));
+
+                db.SaveChanges();
```
Ditto.

randomdavis committed:
Adds positions to opening entity and contract
Adds checking for an incorrect number of saves to a function in the opening service

randomdavis committed:
Removes redundant saving from opening service tests

randomdavis committed:
Removes more redundant saves
